### PR TITLE
Integrate Meerkat into Castanets for Android

### PR DESCRIPTION
**Pull request by younghajung**
This patch integrates Meekat server into Android apk of Castanets.
Meerkat server runs on seperated process of Castanets as foreground
service.
When Castanets is launched first time via app icon, Meerkat server will
be started.

Signed-off-by: yh106.jung <yh106.jung@samsung.com>

### CODE REVIEW

**Comment by younghajung** [comment]
Patch has been rebased on #258.

File: third_party/meerkat/Component/mmSH/android/server_runner_jni.cpp [hunk 1]
```diff
@@ -72,10 +78,11 @@ jint Native_startServer(JNIEnv* env, jobject /* this */) {
   }
 
   ServerRunner::ServerRunnerParams params;
+  // TODO(yh106.jung): Read from configuration file
   params.multicast_addr = "224.1.1.11";
   params.multicast_port = 9901;
   params.service_port = 9902;
-  params.exec_path = "/opt/google/chrome/chrome";
+  params.exec_path = "org.chromium.chrome";
```
Comment (Chikdolman) [inline third_party/meerkat/Component/mmSH/android/server_runner_jni.cpp]: Now we can modify all org.chromium.chrome -> com.samsung.android.castanets

File: third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatServerService.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    @Override
+    public void onDestroy() {
+        if (mMainThread != null) {
+            nativeStopServer();
+            mMainThread = null;
+        }
+        super.onDestroy();
+    }
+
+    public static boolean startChromeRenderer(String args) {
```
Comment (Chikdolman) [inline third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatServerService.java]: How about startCastanetsRenderer?

**Review by djmixkim — APPROVED** [review]
LGTM

File: third_party/meerkat/Component/mmSH/android/server_runner_jni.cpp (hunk 1 above)
Comment (younghajung) [reply third_party/meerkat/Component/mmSH/android/server_runner_jni.cpp]: Done.

File: third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatServerService.java (hunk 2 above)
Comment (younghajung) [reply third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatServerService.java]: Done.

**Comment by suyambulingamrm** [comment]
@younghajung I face below error when launching castanets on android. The app could not get launched. I think its after this change.
Is there any prerequisite to be performed.. ?

jadelte:/ # logcat | grep -i meerkat                                                                                                                                    
11-29 16:26:21.345   997  1153 I ActivityManager: Start proc 7057:com.samsung.android.castanets:meerkat_server_process/u0a466 for service com.samsung.android.castanets/com.samsung.android.meerkat.MeerkatServerService
11-29 16:26:21.351  7057  7057 I SELinux : SELinux: seapp_context_lookup: seinfo=untrusted, level=s0:c512,c768, pkgname=com.samsung.android.castanets:meerkat_server_process 
11-29 16:26:21.476  7057  7057 E AndroidRuntime: Process: com.samsung.android.castanets:meerkat_server_process, PID: 7057
11-29 16:26:21.476  7057  7057 E AndroidRuntime: java.lang.UnsatisfiedLinkError: dalvik.system.PathClassLoader[DexPathList[[zip file "/data/app/com.samsung.android.castanets-1/base.apk"],nativeLibraryDirectories=[/data/app/com.samsung.android.castanets-1/lib/arm, /data/app/com.samsung.android.castanets-1/base.apk!/lib/armeabi-v7a, /system/lib, /vendor/lib]]] couldn't find "libmeerkat_server_lib.so"
11-29 16:26:21.476  7057  7057 E AndroidRuntime: 	at com.samsung.android.meerkat.MeerkatServerService.<clinit>(PG:37)

**Comment by younghajung** [comment]
@suyambulingamrm After this PR, there is no way to launch castanets manually.
Please disable below code in order to launch castanets manually.
`if (startMeerkatServerServiceIfNeeded()) {
            return Action.FINISH_ACTIVITY_REMOVE_TASK;
        }`
I'll find another way to launch castanets manually soon.

**Comment by suyambulingamrm** [comment]
Thanks for the info. I am able to do manual launch after the change.

**Pull request merged**